application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -185,20 +201,35 @@ private void deleteTag(@NotNull CommandInteraction event) {
      *
      * @param requiredTagStatus the required status of the tag
      * @param idAction the action to perform on the id
-     * @param actionVerb the verb describing the executed action, i.e. <i>edited</i> or
-     *        <i>created</i>, will be displayed in the message send to the user
      * @param event the event to send messages with, it must have an {@code id} option set
+     * @param subcommand the executed subcommand
+     * @param newContent the new content of the tag, or null if content is unchanged
      */
     private void handleAction(@NotNull TagStatus requiredTagStatus,
-            @NotNull Consumer<? super String> idAction, @NotNull String actionVerb,
-            @NotNull CommandInteraction event) {
+            @NotNull Consumer<? super String> idAction, @NotNull CommandInteraction event,
+            @NotNull Subcommand subcommand, @Nullable String newContent) {
+
         String id = Objects.requireNonNull(event.getOption(ID_OPTION)).getAsString();
         if (isWrongTagStatusAndHandle(requiredTagStatus, id, event)) {
             return;
         }
 
+        String previousContent = null;
+        if (subcommand == Subcommand.EDIT || subcommand == Subcommand.DELETE) {
+            try {
+                previousContent = tagSystem.getTag(id).orElseThrow();
+            } catch (NoSuchElementException exception) {
+                logger.debug(
```

Comment:
This is in need of a `// NOTE` explaining when this can happen. for example `// NOTE Rare race condition, for example if another thread deleted the tag in the meantime`

Reply:
added it.